### PR TITLE
Adding config service extended implementation

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Implementation
Engaging extended config service impl with registry based implementation.

### CODE REVIEW

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/APIUtil.java [hunk 1]
```diff
@@ -106,6 +106,15 @@ public final  class APIUtil {
...
     private static final int timeoutInSeconds = 15;
     private static final int retries = 2;
+    private static boolean disabledExtendedAPIMConfigService = false;
+
+    public static boolean isDisabledExtendedAPIMConfigService() {
+        return disabledExtendedAPIMConfigService;
+    }
+
+    public static void setDisabledExtendedAPIMConfigService(boolean isDisabledExtendedAPIMConfigService) {
+        APIUtil.disabledExtendedAPIMConfigService = isDisabledExtendedAPIMConfigService;
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/APIUtil.java:116]: @CrowleyRajapakse do we need to explictly refer disabledExtendedAPIMConfigService as APIUtl.disabledExtendedAPIMConfigService if in the same class?

Comment (CrowleyRajapakse) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/APIUtil.java]: changed and fixed.

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import javax.cache.Cache;
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+
+public class ExtendedAPIMConfigServiceImpl extends APIMConfigServiceImpl {
+
+    @Override
+    public void addTenantConfig(String organization, String tenantConfig) throws APIManagementException {
+
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java:26]: @CrowleyRajapakse shall we add more info logs on both if else paths for debugging purposes.

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V410Migration.java [hunk 3]
```diff
@@ -42,5 +43,6 @@ public void migrate() throws UserStoreException, APIMigrationException {
         postDBScriptMigratorForRegDB.run();
         PostDBScriptMigrator postDBScriptMigratorForAmDb = new PostDBScriptMigrator(POST_MIGRATION_SCRIPT_AMDB_PATH);
         postDBScriptMigratorForAmDb.run();
+        APIUtil.setDisabledExtendedAPIMConfigService(true);
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V410Migration.java:46]: lets add info logs here as well as we are disabling the service.

Comment (CrowleyRajapakse) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/client/V410Migration.java:48]: fixed

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            }
+            Cache tenantConfigCache = CacheProvider.getTenantConfigCache();
+            String cacheName = organization + "_" + APIConstants.TENANT_CONFIG_CACHE_NAME;
+            tenantConfigCache.remove(cacheName);
+            try {
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(organization, true);
+                int tenantId = APIUtil.getTenantIdFromTenantDomain(organization);
+                if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(organization)) {
+                    APIUtil.loadTenantRegistry(tenantId);
```
Comment (tharindu1st) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java:120]: please duplicate this method in migration client

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            }
+            try {
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(organization, true);
+                int tenantId = APIUtil.getTenantIdFromTenantDomain(organization);
+                if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(organization)) {
+                    APIUtil.loadTenantRegistry(tenantId);
+                }
+
+                RegistryService registryService = ServiceReferenceHolder.getInstance().getRegistryService();
```
Comment (tharindu1st) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java:46]: Don't use ServiceReferenceHolder from apimgt.impl

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java (hunk 4 above)
Comment (CrowleyRajapakse) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java:119]: Resolved, changed the impl APIUtil import to migration util class which already had this duplicated method.

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java (hunk 5 above)
Comment (CrowleyRajapakse) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/util/ExtendedAPIMConfigServiceImpl.java]: changed, resolved.

**Review by dushaniw — APPROVED** [review]

**Pull request merged**